src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminal.accessibility.contribution.ts

```diff
@@ -90,23 +93,35 @@ export class TerminalAccessibleViewContribution extends Disposable implements IT
 		xterm.raw.loadAddon(addon);
 		addon.activate(xterm.raw);
 		this._xterm = xterm;
+		this._register(this._xterm.raw.onWriteParsed(async () => {
+			if (this._isTerminalAccessibleViewOpen() && this._xterm!.raw.buffer.active.baseY === 0) {
+				this.show();
+			}
+		}));
+
+		const onRequestUpdateEditor = Event.latch(this._xterm.raw.onScroll);
```

Comment:
You need to register the new `Event` created via `Event.latch`. I think best practice would be:

```ts
		this._register(Event.latch(this._xterm.raw.onScroll, this._store)(() => {
			if (this._isTerminalAccessibleViewOpen()) {
				this.show();
			}
		}));
```

Reply:
Ok, is that different from registering the const that gets created as is currently done on the line below that one?

Reply:
Actually it might be find actually. It's a little tough to understand but I think since `Emitter` is not a `DisposableStore`, disposing of it is optional. Its dispose just unsets everything:

https://github.com/microsoft/vscode/blob/613640d25feb7b9bdd8c6d056a7d5c7acfa870f8/src/vs/base/common/event.ts#L988-L1019

This is why this note exists:

https://github.com/microsoft/vscode/blob/613640d25feb7b9bdd8c6d056a7d5c7acfa870f8/src/vs/base/common/event.ts#L323-L329

If listeners exist outside of a class, ie. the event listeners created outside may not be disposed when Event.latch's Emitter is implicitly disposed, they can hold on to references.

All good as is 👌